### PR TITLE
executor, privilege: fix failure on grant USAGE privilege operation

### LINKED ISSUE #22721: grant USAGE privilege operation fail
## Bug Report

Please answer these questions before submitting your issue. Thanks!

### 1. Minimal reproduce step (Required)

<!-- a step by step guide for reproducing the bug. -->

```
CREATE USER 'sync_ci_data'@'%' IDENTIFIED BY 'sNGNQo12fEHe0n3vU';

mysql> GRANT USAGE ON sync_ci_data.* TO 'sync_ci_data'@'%';
ERROR 8121 (HY000): privilege check fail
```

### 2. What did you expect to see? (Required)

`GRANT USAGE ON xx TO xx` should not return error.

### 3. What did you see instead (Required)

```
ERROR 8121 (HY000): privilege check fail
```

### 4. What is your TiDB version? (Required)

<!-- Paste the output of SELECT tidb_version() -->

```
mysql> select tidb_version();
+---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
| tidb_version()                                                                                                                                                                                                                                                                                                                  |
+---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
| Release Version: v4.0.0-beta.2-2092-gee876397a-dirty
Edition: Community
Git Commit Hash: ee876397ac020a6b164d56b4a448b61c4641acf3
Git Branch: master
UTC Build Time: 2021-02-03 07:47:52
GoVersion: go1.15
Race Enabled: false
TiKV Min Version: v3.0.0-60965b006877ca7234adaced7890d7b029ed1306
Check Table Before Drop: false |
+---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
1 row in set (0.03 sec)
```

4.0, 5.0, master(b35e15c8cfeaa7e8f4d8a0db1c6cbba88ff116d9)

### PR DESCRIPTION
**Pull request by TszKitLo40**
### What problem does this PR solve?

Issue Number: close #22721  

Problem Summary:

### What is changed and how it works?

What's Changed:
 fix failure on grant USAGE privilege operation
How it Works:
treat `mysql.UsagePriv` as nop operation
### Check List 

Tests 

- Unit test
- Integration test

Side effects

- Performance regression
    - Consumes more CPU
    - Consumes more MEM
- Breaking backward compatibility

### Release note 

- No release note

### CODE REVIEW

**Comment by morgo** [comment]
LGTM

File: store/helper/helper.go [hunk 1]
```diff
@@ -587,7 +603,13 @@ func (h *Helper) GetRegionsTableInfo(regionsInfo *RegionsInfo, schemas []*model.
 	tables := []tableInfoWithKeyRange{}
 	for _, db := range schemas {
 		for _, table := range db.Tables {
-			tables = append(tables, newTableWithKeyRange(db, table))
+			if table.Partition != nil {
```
Comment (bb7133) [inline store/helper/helper.go:606]: Why changing the codes here? Looks that it is not related to the 'privileges'.

Comment (TszKitLo40) [reply store/helper/helper.go:606]: I will remove the code here. Seems that it is caused by I checkout with the other branch.

Comment (TszKitLo40) [reply store/helper/helper.go]: I have removed it. PTAL

**Pull request merged**